src/main/java/org/mule/extension/db/internal/operation/BulkOperations.java

```diff
@@ -195,9 +197,17 @@ private Query resolveQuery(BulkQueryDefinition query,
   }
 
   private List<List<QueryParamValue>> resolveParamSets(List<Map<String, Object>> values) {
-    return values.stream().map(map -> map.entrySet().stream()
-        .map(entry -> new QueryParamValue(entry.getKey(), entry.getValue()))
-        .collect(toList()))
-        .collect(toList());
+    List<List<QueryParamValue>> parameterSet = new ArrayList<>();
+    for (Object value : values) {
+      Map<String, Object> map;
+      if (value instanceof TypedValue) {
+        map = TypedValue.unwrap(((TypedValue) value).getValue());
+      } else {
+        map = (Map<String, Object>) value;
+      }
```

Comment:
This code could be improved by always calling the static  unwrap method. The instance of check is already done inside.

Reply:
Also, the import should be of the static method